fiftyone/utils/flash.py

```diff
@@ -0,0 +1,93 @@
+"""
+PyTorch Lightning Flash utilities.
+
+| Copyright 2017-2021, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+import logging
+
+import fiftyone.core.labels as fol
+import fiftyone.core.utils as fou
+
+fou.ensure_lightning_flash()
+import flash
+from flash.core.classification import (
+    ClassificationSerializer,
+    ClassificationTask,
+    FiftyOneLabels,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+_SUPPORTED_TASKS = [
+    ClassificationTask,
+]
+
+
+def apply_flash_model(
+    samples,
+    model,
+    label_field="predictions",
+    confidence_thresh=None,
+    store_logits=False,
+    batch_size=None,
+    num_workers=None,
+    skip_failures=True,
+):
+    """Applies the given ``flash.core.model.Task`` to the samples in the 
+    collection.
+
+    This method supports all the following cases:
+
+    -   Applying an image model to an image collection
+    -   Applying an image model to the frames of a video collection
+    -   Applying a video model to a video collection
+
+    Args:
+        samples: a :class:`fiftyone.core.collections.SampleCollection`
+        model: a ``flash.core.model.Task``
+        label_field ("predictions"): the name of the field in which to store
+            the model predictions. When performing inference on video frames,
+            the "frames." prefix is optional
+        confidence_thresh (None): an optional confidence threshold to apply to
+            any applicable labels generated by the model
+        store_logits (False): whether to store logits for the model
+            predictions. This is only supported when the provided ``model`` has
+            logits
+        batch_size (None): an optional batch size to use. Only applicable for
+            image samples
+        num_workers (None): the number of workers to use when loading images.
+            Only applicable for Torch models
+        skip_failures (True): whether to gracefully continue without raising an
+            error if predictions cannot be generated for a sample
+    """
+    serializer = _get_fo_serializer(model, confidence_thresh)
+    model.serializer = serializer
+    with fou.ProgressBar() as pb:
```

Comment:
I haven't investigated in great detail, but shouldn't Flash provide support for running prediction on an entire dataset via a single call to `predict()` that would then use their fancy data loading + distributed setup to perform the inference?

In other words, shouldn't you instead use the syntax developed in https://github.com/voxel51/lightning-flash/pull/2 here?

This may be a question for the Flash team if you can't find anything on your own.

Reply:
So I've updated this to just use `predict()`:
https://github.com/voxel51/fiftyone/blob/f8a65320b2ba23ec070f69bd3b2a5f5b294d0bde/fiftyone/utils/flash.py#L72-L76

The only issue now is that there is no progress bar, I guess we just have to live with that?